k-es-lib/src/test/kotlin/no/ks/kes/lib/SagaConfigurationTest.kt

```diff
@@ -0,0 +1,99 @@
+package no.ks.kes.lib
+
+import io.kotlintest.matchers.string.shouldContain
+import io.kotlintest.shouldThrow
+import io.kotlintest.specs.StringSpec
+import java.time.Instant
+import java.util.*
+
+class SagaConfigurationTest : StringSpec() {
+    private data class SomeAggregate(val stateInitialized: Boolean, val stateUpdated: Boolean = false) : Aggregate
+
+    private data class SomeState(val id: UUID)
+
+    @SerializationId("some-id")
+    private data class SomeEvent(override val aggregateId: UUID, override val timestamp: Instant) : Event<SomeAggregate>
+
+    @Deprecated(message = "dont use this event")
+    @SerializationId("some-deprecated-event")
+    private data class SomeDeprecatedEvent(override val aggregateId: UUID, override val timestamp: Instant) : Event<SomeAggregate>
+
+    init {
+
+        "test that creating a saga with multiple initializers throws exception" {
+            shouldThrow<IllegalStateException> {
+                object : Saga<SomeState>(SomeState::class, "SomeSaga") {
+                    init {
+                        init<SomeEvent>({ it.aggregateId }) { SomeState(it.aggregateId) }
+                        init<SomeEvent>({ it.aggregateId }) { SomeState(it.aggregateId) }
+                    }
+                }
+            }.apply {
```

Comment:
May be simplified. Why not do something like:
```kotlin
shouldThrow<IllegalStateException> {
...
}.message shouldContain "Duplicate init handler" 
```

The use of an apply function sort of implies that you are making some change to the state of the object which you do not here (also used in other tests)

Reply:
agreed, scoping func not necessary as long as only one assert is executed, and even then apply is probably not the best choice: "run" is better maybe?

Reply:
Yes "run" would be better if you have more than one assertion